src/openrct2/world/Sprite.h

```diff
@@ -193,15 +193,21 @@ enum
 
 rct_sprite* try_get_sprite(size_t spriteIndex);
 rct_sprite* get_sprite(size_t sprite_idx);
-template<typename T> T* GetEntity(size_t sprite_idx)
+template<typename T = SpriteBase> T* GetEntity(size_t sprite_idx)
 {
     auto spr = reinterpret_cast<SpriteBase*>(get_sprite(sprite_idx));
     if (spr == nullptr)
         return nullptr;
     return spr->As<T>();
 }
 
-SpriteBase* GetEntity(size_t sprite_idx);
+template<typename T = SpriteBase> T* TryGetEntity(size_t sprite_idx)
+{
+    auto spr = reinterpret_cast<SpriteBase*>(try_get_sprite(sprite_idx));
+    if (spr == nullptr)
+        return nullptr;
+    return spr->As<T>();
```

Comment:
```suggestion
    return spr == nullptr ? spr->As<T>() : nullptr;
```
I'd shorten it, but your call